packages/dashmate/src/helper/renewZeroSslCertificateHelperFactory.js

```diff
@@ -0,0 +1,62 @@
+const { CronJob } = require('cron');
+
+/**
+ *
+ * @param {listCertificates} listCertificates
+ * @param {renewZeroSSLCertificateTask} renewZeroSSLCertificateTask
+ * @param {DockerCompose} dockerCompose
+ * @return {renewZeroSslCertificateHelper}
+ */
+function renewZeroSslCertificateHelperFactory(
+  listCertificates,
+  renewZeroSSLCertificateTask,
+  dockerCompose,
+) {
+  /**
+   * @typedef renewZeroSslCertificateHelper
+   * @param {Config} config
+   * @return {Promise<void>}
+   */
+  async function renewZeroSslCertificateHelper(config) {
+    const provider = config.get('platform.dapi.envoy.ssl.provider');
+    if (provider !== 'zerossl') {
+      return;
+    }
+
+    const certificatesResponse = await listCertificates(config.get('platform.dapi.envoy.ssl.providerConfigs.zerossl.apiKey'));
+
+    const certificate = certificatesResponse.results.find((result) => result.common_name === config.get('externalIp'));
```

Comment:
Better to store certificate id to config when you obtain it and use this id here